internal/gatewayapi/backendtlspolicy.go

```diff
@@ -157,13 +157,54 @@ func backendTLSTargetMatched(policy gwapiv1a3.BackendTLSPolicy, target gwapiv1a2
 	return false
 }
 
-func getBackendTLSPolicy(policies []*gwapiv1a3.BackendTLSPolicy, backendRef gwapiv1a2.BackendObjectReference, backendNamespace string) *gwapiv1a3.BackendTLSPolicy {
+// getTargetBackendReferenceWithPortName returns the LocalPolicyTargetReference for the given BackendObjectReference,
+// and sets the sectionName to the port name if the BackendObjectReference is a Kubernetes Service.
+func getTargetBackendReferenceWithPortName(
+	backendRef gwapiv1a2.BackendObjectReference,
+	backendNamespace string,
+	resources *resource.Resources,
+) gwapiv1a2.LocalPolicyTargetReferenceWithSectionName {
+	ref := getTargetBackendReference(backendRef)
+	if ref.SectionName == nil {
+		return ref
+	}
+	if backendRef.Kind != nil && *backendRef.Kind != resource.KindService {
+		return ref
+	}
+
+	if service := resources.GetService(backendNamespace, string(backendRef.Name)); service != nil {
+		for _, port := range service.Spec.Ports {
+			if port.Port == int32(*backendRef.Port) {
+				if port.Name != "" {
+					ref.SectionName = SectionNamePtr(port.Name)
+				}
+			}
+		}
+	}
+
+	return ref
+}
+
+func getBackendTLSPolicy(
+	policies []*gwapiv1a3.BackendTLSPolicy,
+	backendRef gwapiv1a2.BackendObjectReference,
+	backendNamespace string,
+	resources *resource.Resources,
+) *gwapiv1a3.BackendTLSPolicy {
 	target := getTargetBackendReference(backendRef)
 	for _, policy := range policies {
 		if backendTLSTargetMatched(*policy, target, backendNamespace) {
 			return policy
 		}
 	}
+
+	// SectionName can be port name for Kubernetes Service
+	target = getTargetBackendReferenceWithPortName(backendRef, backendNamespace, resources)
```

Comment:
why do we need another new function
isnt there logic already for this in https://github.com/envoyproxy/gateway/blob/1536c2a3538da9f6322351f5276c0453a7a34e24/internal/gatewayapi/backendtlspolicy.go#L149 ?

Reply:
This function uses port name instead of port number as sectionName to match policy.

Reply:
lets support port number - section name matching for Backend and only port name - section name matching for Service, this ensures the user config can be safely migrated and tested in other controllers